Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py

```diff
@@ -136,36 +161,64 @@ def create_values_out_dict(iocs: list, out_format: str) -> dict:
 
 def get_outbound_mimetype() -> str:
     """Returns the mimetype of the export_iocs"""
-    ctx = demisto.getIntegrationContext()
+    ctx = demisto.getIntegrationContext().get('last_output')
     return ctx.get(CTX_MIMETYPE_KEY, 'text/plain')
 
 
-def get_outbound_ioc_values(on_demand, limit, indicator_query='', out_format='text', last_run=None,
+def get_outbound_ioc_values(on_demand, limit, offset, indicator_query='', out_format=FORMAT_TEXT, last_update_data={},
                             cache_refresh_rate=None) -> str:
     """
     Get the ioc list to return in the list
     """
+    last_update = last_update_data.get('last_run')
+    last_limit = last_update_data.get('last_limit')
+    last_offset = last_update_data.get('last_offset')
+    last_format = last_update_data.get('last_format')
+    last_query = last_update_data.get('last_query')
+    current_iocs = last_update_data.get('current_iocs')
+
     # on_demand ignores cache
     if on_demand:
-        values_str = get_ioc_values_str_from_context()
+        if out_format != last_format or limit != last_limit or offset != last_offset:
+            values_str = get_ioc_values_str_from_context(current_iocs, out_format, limit, offset)
+
+        else:
+            values_str = get_ioc_values_str_from_context()
+
     else:
-        if last_run:
+        if last_update:
+            # takes the cache_refresh_rate amount of time back since run time.
             cache_time, _ = parse_date_range(cache_refresh_rate, to_timestamp=True)
-            if last_run <= cache_time:
-                values_str = refresh_outbound_context(indicator_query, out_format, limit=limit)
+            if last_update <= cache_time or last_limit != limit or \
+                    last_offset != offset or last_format != out_format or indicator_query != last_query:
```

Comment:
Consider extracting this along with lines 182-183 to a boolean function such as `is_request_change`, it might make the code a bit longer, however, it'll make it easier to understand what you're checking here, and to test this logic. Please make sure to add a unit test for this logic condition.